uniond/app/upgrades/v1_2_0/upgrade.go

```diff
@@ -0,0 +1,240 @@
+	return func(ctx context.Context, plan upgradetypes.Plan, vm module.VersionMap) (module.VersionMap, error) {
+		migrations, err := mm.RunMigrations(ctx, configurator, vm)
+		if err != nil {
+			return nil, err
+		}
+
+		unionFoundationMultiSig, err := sdk.AccAddressFromBech32(UNION_FOUNDATION_MULTI_SIG)
+		if err != nil {
+			return nil, err
+		}
+
+		// Undelegate existing delegations
+		delegations, err := keepers.StakingKeeper.GetAllDelegations(ctx)
+		if err != nil {
+			return nil, err
+		}
+
+		for _, delegation := range delegations {
+			accAddr, err := sdk.AccAddressFromBech32(delegation.DelegatorAddress)
+			if err != nil {
+				return nil, err
+			}
+			valAddr, err := sdk.ValAddressFromBech32(delegation.ValidatorAddress)
+			if err != nil {
+				return nil, err
+			}
+
+			_, err = keepers.DistributionKeeper.WithdrawDelegationRewards(ctx, accAddr, valAddr)
+			if err != nil {
+				return nil, err
+			}
+			_, err = keepers.DistributionKeeper.WithdrawValidatorCommission(ctx, valAddr)
+			if err != nil {
+				return nil, err
+			}
+
+			validator, err := keepers.StakingKeeper.GetValidator(ctx, valAddr)
+			if err != nil {
+				return nil, err
+			}
+			validator.MinSelfDelegation = math.NewInt(0)
+			keepers.StakingKeeper.SetValidator(ctx, validator)
+
+			_, _, err = keepers.StakingKeeper.Undelegate(ctx, accAddr, valAddr, delegation.Shares)
+			if err != nil {
+				return nil, err
+			}
+		}
+
+		// Mint U
+		uTotalSupply := []sdk.Coin{getUFromU64(U_TOTAL_SUPPLY)}
+		err = keepers.MintKeeper.MintCoins(ctx, uTotalSupply)
+		if err != nil {
+			return nil, err
+		}
+		err = keepers.BankKeeper.SendCoinsFromModuleToAccount(ctx, minttypes.ModuleName, unionFoundationMultiSig, uTotalSupply)
+		if err != nil {
+			return nil, err
+		}
+
+		// Update x/staking
+		stakingParams, err := keepers.StakingKeeper.GetParams(ctx)
+		if err != nil {
+			return nil, err
+		}
+		stakingParams.BondDenom = U_BASE_DENOM
+		stakingParams.MinCommissionRate = math.LegacyMustNewDecFromStr("0.05")
+		err = keepers.StakingKeeper.SetParams(ctx, stakingParams)
+		if err != nil {
+			return nil, err
+		}
+
+		// Update x/mint
+		// NOTE: Keeping inflation set to 0 until TGE
+		mintparams, err := keepers.MintKeeper.Params.Get(ctx)
+		if err != nil {
+			return nil, err
+		}
+		mintparams.MintDenom = U_BASE_DENOM
+		err = keepers.MintKeeper.Params.Set(ctx, mintparams)
+		if err != nil {
+			return nil, err
+		}
+
+		// Update x/gov
+		govParams, err := keepers.GovKeeper.Params.Get(ctx)
+		if err != nil {
+			return nil, err
+		}
+		govParams.MinDeposit = []sdk.Coin{getUFromU64(10)}
+		govParams.ExpeditedMinDeposit = []sdk.Coin{getUFromU64(50)}
+		err = keepers.GovKeeper.Params.Set(ctx, govParams)
+		if err != nil {
+			return nil, err
+		}
+
+		// Update x/crisis
+		_, err = keepers.CrisisKeeper.UpdateParams(ctx, &crisistypes.MsgUpdateParams{
+			Authority:   keepers.GovKeeper.GetAuthority(),
+			ConstantFee: getBaseUFromString("1000"),
```

Comment:
where did this value come from/ what does it mean?

Reply:
This is what it's set to on both networks genesis.

> Due to the anticipated large gas cost requirement to verify an invariant (and potential to exceed the maximum allowable block gas limit) a constant fee is used instead of the standard gas consumption method. The constant fee is intended to be larger than the anticipated gas cost of running the invariant with the standard gas consumption method.

We can update it to be higher if need be.

Reply:
also why not just use getUFromU64 here lol